python/python/benchmarks/test_v2_reader.py

```diff
@@ -0,0 +1,219 @@
+import csv
+import os
+import shutil
+from datetime import datetime
+
+import pyarrow as pa
+import pyarrow.dataset as ds
+import pyarrow.parquet as pq
+from lance.file import LanceFileReader, LanceFileWriter
+from lance.tracing import trace_to_chrome
+
+trace_to_chrome(level="trace")
+
+"""
+Data generation (in-memory)
+import duckdb
+import pyarrow.parquet as pq
+
+con = duckdb.connect(database=':memory:')
+con.execute("INSTALL tpch; LOAD tpch")
+con.execute("CALL dbgen(sf=10)")
+
+tables = ["lineitem"]
+for t in tables:
+    res = con.query("SELECT * FROM " + t)
+    pq.write_table(res.to_arrow_table(), t + ".parquet")
+"""
+
+
+def write_lance_file(dataset, dir_path="/tmp/tpch.lancev2"):
+    """
+    Takes a pyarrow dataset object as input, and writes a lance
+    file to disk in a directory
+    """
+
+    with LanceFileWriter(dir_path, dataset.schema) as writer:
+        for batch in dataset.to_batches():
+            writer.write_batch(batch)
+
+
+def measure_pyarrow_read_time(path, num_trials, verbose=False):
+    """
+    Measures the time required to read a parquet file using pyarrow,
+    averaged over multiple trials
+    """
+    parquet_read_time = 0
+    for trial in range(1, num_trials + 1):
+        start = datetime.now()
+        pa.parquet.read_table(path)
+        end = datetime.now()
+        parquet_read_time += (end - start).total_seconds()
+
+        if verbose:
+            print(f"Parquet Read Time: {parquet_read_time/trial}s")
```

Comment:
instead of the average read_time so far, maybe the read_time for this particular trial is better? like `print(f"Parquet Read Time for trial {trial}: {(end - start).total_seconds}s")`